### ISSUE TITLE
Missing status when sbr reconcile fails

### ISSUE THREAD

**Issue by cdlliuy** [redhat-developer/service-binding-operator#413]
Not sure whether it is a duplication, but  I didn't see  the sbr status field at all when the reconcile fails..

Also, an error msg field is welcome in the `status` field, so that the end-user can understand what is the error in the CR yaml.

**Comment by sbose78** [comment]
This probably broke recently https://github.com/redhat-developer/service-binding-operator/issues/408 cc @DhritiShikhar ?

**Comment by DhritiShikhar** [comment]
@sbose78 Yes. I have raised a PR for the same.

**Comment by Avni-Sharma** [comment]
Hi @cdlliuy . Thanks for the issue. The issue is tracked at #408 and fix for the same is being tracked here #414

**Comment by DhritiShikhar** [comment]
@Avni-Sharma This issue is not completely solved. 

Can you re-open this issue?

**Comment by Avni-Sharma** [comment]
@DhritiShikhar If we omit backingServiceSelector, then it was previously handled here https://github.com/redhat-developer/service-binding-operator/blob/35e8d3a1988e74880c4d15a35698ab154b6162f9/pkg/controller/servicebindingrequest/planner.go#L95  already and so it shows something like 
```
status:
  bindingStatus: ""
  conditions:
  - lastHeartbeatTime: "2020-04-07T16:06:16Z"
    lastTransitionTime: "2020-04-07T16:06:16Z"
    message: backing service selectors are empty
    reason: EmptyBackingServiceSelectors
    status: "False"
    type: Ready
  secret: ""

```
On empty application selector it shows as 
```
status:
  bindingStatus: ""
  conditions:
  - lastHeartbeatTime: "2020-04-07T10:48:29Z"
    lastTransitionTime: "2020-04-07T10:48:28Z"
    message: application ResourceRef or MatchLabel not found
    reason: EmptyApplicationSelector
    status: "False"
    type: Ready
  secret: ""
```

if we put an empty spec as {}
then as applicationselector is tackled first in the if case so it updates the status as well with the same as the ApplicationSelector absence.
So currently the status is being updated on reconcile. Can you reproduce this issue, if it exists still?

**Comment by cdlliuy** [comment]
@DhritiShikhar  is PR #414 enough for this issue? or you will add another PR to address further probolem described in this issue?

**Comment by DhritiShikhar** [comment]
This issue is fixed in #407
